chain/network/src/peer/peer_actor.rs

```diff
@@ -418,6 +419,11 @@ impl PeerActor {
     fn stop(&mut self, ctx: &mut actix::Context<PeerActor>, reason: ClosingReason) {
         // Only the first call to stop sets the closing_reason.
         if self.closing_reason.is_none() {
+            // Send a Disconnect message unless we are closing because we received one
+            if reason != ClosingReason::DisconnectMessage {
+                self.send_message_or_log(&PeerMessage::Disconnect(reason.clone()));
+            }
```

Comment:
do you really need to make it conditional? AFAICT peer just won't receive the disconnect message, which would WAI, right?